lib/MemoryHandler.js

```diff
@@ -133,3 +157,7 @@ MemoryStore.prototype._sortList = function(request, list) {
     }
   });
 };
+
+MemoryStore.prototype._clone = function(obj) {
+  return JSON.parse(JSON.stringify(obj));
```

Comment:
This data is intended to be serialised at this point for storage. If anything does have a circular reference at this point, it's going to break for real when the memory handler gets replaced for a real storage mechanism?